File: rb/features.rb
# UuidGenerator SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module UuidGeneratorFeatures
  def self.make_feature(name)
    case name
    when "base"
      UuidGeneratorBaseFeature.new
    when "test"
      UuidGeneratorTestFeature.new
    else
      UuidGeneratorBaseFeature.new
    end
  end
end

File: py/uuidgenerator_sdk/features.py
# UuidGenerator SDK feature factory

from uuidgenerator_sdk.feature.base_feature import UuidGeneratorBaseFeature
from uuidgenerator_sdk.feature.test_feature import UuidGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UuidGeneratorBaseFeature(),
        "test": lambda: UuidGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
